library/core/registry/src/main/java/org/quiltmc/qsl/registry/impl/RegistryConfig.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2022 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qsl.registry.impl;
+
+import org.jetbrains.annotations.ApiStatus;
+import org.quiltmc.config.api.Config;
+import org.quiltmc.config.api.values.TrackedValue;
+import org.quiltmc.loader.api.config.QuiltConfig;
+
+@ApiStatus.Internal
+public class RegistryConfig {
+	public static final String REGISTRY_SYNC = "registry_sync";
+	public static final String NO_REGISTRY_SYNC_MESSAGE = "no_registry_sync_message";
+	public static final String SUPPORT_FABRIC_API_SYNC = "support_fabric_api_sync";
+	private static Config config = null;
+	public static Config getConfig() {
```

Comment:
The way that the config system is implemented is way too cursed; I'd recommend basing it on Ok Zoomer's [Config](https://github.com/EnnuiL/OkZoomer/blob/1.19/src/main/java/io/github/ennuil/ok_zoomer/config/OkZoomerConfig.java)/[ConfigManager](https://github.com/EnnuiL/OkZoomer/blob/1.19/src/main/java/io/github/ennuil/ok_zoomer/config/OkZoomerConfigManager.java) system, or even Haven's [basic Config example](https://github.com/Haven-King/developer-wiki/blob/main/wiki/configs/basic/markdown/basic.md), considering that the only thing being done with the config entries is reading it
For keeping snake_case on these examples (which I do recommend, please don't use camelCase on TOML), there's still the cursedness of using it on few fields, but we can fix it later once naming conventions are here